docs/docs/reference/augments/method-augments.md

```diff
@@ -0,0 +1,189 @@
+---
+layout: doc-page
+title: "Method Augmentations"
+---
+
+Method augmentations are a way to define _extension methods_. Here is a simple one:
+
+```scala
+case class Circle(x: Double, y: Double, radius: Double)
+
+augment Circle {
+  def circumference: Double = this.radius * math.Pi * 2
+}
+```
+
+The `augment Circle` clause adds an extension method `circumference` to values of class `Circle`. Like regular methods, extension methods can be invoked with infix `.`:
+
+```scala
+  val circle = Circle(0, 0, 1)
+  circle.circumference
+```
+
+### Meaning of `this`
+
+Inside an extension method, the name `this` stands for the receiver on which the
+method is applied when it is invoked. E.g. in the application of `circle.circumference`,
+the `this` in the body of `circumference` refers to `circle`. Unlike for regular methods,
+an explicit `this` is mandatory to refer to members of the receiver. So the following
+gives a compilation error:
+
+```scala
+   |    def circumference = radius * math.Pi * 2
+   |                        ^^^^^^
+   |                        not found: radius
+```
+
+### Scope of Augment Clauses
+
+Augment clauses can appear anywhere in a program; there is no need to co-define them with the types they augment. Extension methods are available whereever their defining augment clause is in scope.  Augment clauses can be inherited or wildcard-imported like normal definitions. This is usually sufficient to control their visibility. If more control is desired, one can also attach a name to an augment clause, like this:
+
+```scala
+package shapeOps
+
+augment circleOps @ Circle {
```

Comment:
Can we have top level augment clauses or they share the same restriction as implicit classes? If we can, we are missing a test case for it

Reply:
Currently these are not possible.